pkg/scalers/solace_scaler.go

```diff
@@ -0,0 +1,367 @@
+	// Create HTTP Client
+	httpClient := kedautil.CreateHTTPClient(config.GlobalHTTPTimeout)
+
+	// Parse Solace Metadata
+	solaceMetadata, err := parseSolaceMetadata(config)
+	if err != nil {
+		solaceLog.Error(err, "Error parsing Solace Trigger Metadata or missing values")
+		return nil, err
+	}
+
+	return &SolaceScaler{
+		metadata:   solaceMetadata,
+		httpClient: httpClient,
+	}, nil
+}
+
+//	Called by constructor
+func parseSolaceMetadata(config *ScalerConfig) (*SolaceMetadata, error) {
+	meta := SolaceMetadata{}
+	//	GET THE SEMP API ENDPOINT
+	if val, ok := config.TriggerMetadata[solaceMetaSempBaseURL]; ok && val != "" {
+		meta.solaceSempURL = val
+	} else {
+		return nil, fmt.Errorf(solaceFoundMetaFalse, solaceMetaSempBaseURL)
+	}
+	//	GET Message VPN
+	if val, ok := config.TriggerMetadata[solaceMetaMsgVpn]; ok && val != "" {
+		meta.messageVpn = val
+	} else {
+		return nil, fmt.Errorf(solaceFoundMetaFalse, solaceMetaMsgVpn)
+	}
+	//	GET Queue Name
+	if val, ok := config.TriggerMetadata[solaceMetaQueueName]; ok && val != "" {
+		meta.queueName = val
+	} else {
+		return nil, fmt.Errorf(solaceFoundMetaFalse, solaceMetaQueueName)
+	}
+
+	//	GET METRIC TARGET VALUES
+	//	GET msgCountTarget
+	if val, ok := config.TriggerMetadata[solaceMetaMsgCountTarget]; ok && val != "" {
+		if msgCount, err := strconv.Atoi(val); err == nil {
+			meta.msgCountTarget = msgCount
+		} else {
+			return nil, fmt.Errorf("can't parse [%s], not a valid integer: %s", solaceMetaMsgCountTarget, err)
+		}
+	}
+	//	GET msgSpoolUsageTarget
+	if val, ok := config.TriggerMetadata[solaceMetaMsgSpoolUsageTarget]; ok && val != "" {
+		if msgSpoolUsage, err := strconv.Atoi(val); err == nil {
+			meta.msgSpoolUsageTarget = msgSpoolUsage * 1024 * 1024
+		} else {
+			return nil, fmt.Errorf("can't parse [%s], not a valid integer: %s", solaceMetaMsgSpoolUsageTarget, err)
+		}
+	}
+
+	//	Check that we have at least one positive target value for the scaler
+	if meta.msgCountTarget < 1 && meta.msgSpoolUsageTarget < 1 {
+		return nil, fmt.Errorf("no target value found in the scaler configuration")
+	}
+
+	// Format Solace SEMP Queue Endpoint (REST URL)
+	meta.endpointURL = fmt.Sprintf(
+		solaceSempEndpointURLTemplate,
+		meta.solaceSempURL,
+		solaceAPIName,
+		solaceAPIVersion,
+		meta.messageVpn,
+		solaceAPIObjectTypeQueue,
+		meta.queueName)
+
+	// Get Credentials
+	var e error
+	if meta.username, meta.password, e = getSolaceSempCredentials(config); e != nil {
+		return nil, e
+	}
+	return &meta, nil
+}
+
+func getSolaceSempCredentials(config *ScalerConfig) (u string, p string, err error) {
+	//	GET CREDENTIALS
+	//	The username must be a valid broker ADMIN user identifier with read access to SEMP for the broker, VPN, and relevant objects
+	//	The scaler will attempt to acquire username and then password independently. For each:
+	//	- Search K8S Secret (Encoded)
+	//	- Search environment variable specified by config at 'usernameEnv' / 'passwordEnv'
```

Comment:
```suggestion
	//	- Search environment variable specified by config at 'usernameFromEnv' / 'passwordFromEnv'
```
Minor nit, we are using `fromEvn` suffix for this kind of variables

Reply:
🤖 _Did you mean `fromEnv`?_

Reply:
🤦 oh boy...

Reply:
I'll put in a change - also need to update the doc.